src/Core/Repository/AbstractMultiShopObjectModelRepository.php

```diff
@@ -190,6 +190,9 @@ protected function checkShopAssociation(int $id, string $objectModelClassName, S
      */
     protected function deleteObjectModelFromShops(ObjectModel $objectModel, array $shopIds, string $exceptionClass, int $errorCode = 0): void
     {
+        if (empty($shopIds)) {
```

Comment:
Hmm, why would it be empty here? Actually if it is it means something went wrong before so instead of ignoring it I think an exception should be thrown instead

Reply:
I don't know why this array would be empty. Anyway the function allow to pass an empty array so we have to consider it. This line is purely for performance purpose. If shopIds is empty then we don't have any reason to execute the following code.

Reply:
The input shouldn't be empty!! So I agree we can improve this but just ignoring it is not the right IMHO we should instead force the array not to be empty